Update to the helm templates to allow nodeport numbers to be defined in values.

tmojlupo opened the pull request:
Adding logic to 8-gateway-proxy-service.yaml to use nodeport numbers if predefined in the values.yml.  This would satisfy issue #2899.
https://github.com/solo-io/gloo/issues/2899

It can be used as follows in a 
gloo:
  gatewayProxies:
    gatewayProxy:
      service:
        httpPort: 80
        httpNodePort: 30001
        httpsPort: 443
        httpsNodePort: 30002
        type: NodePort
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2899

tmojlupo commented:
> You'll need to add a new changelog file

Added with commit e143ce0.

kdorosh commented:
/test

kdorosh commented:
/test untar mod cache flake, https://console.cloud.google.com/cloud-build/builds/6bb357fd-c244-4dc5-9830-f5b5645080b2?project=solo-public

tmojlupo commented:
I updated the location of the changelog file to v1.4.0-beta9, updated the values.go file for the documentation and ran the "make update-deps generated-code -B" command to generate the updated values.txt document.

kdorosh commented:
/test

kdorosh commented:
/test regression test flake https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=19ab2854-f71b-4a10-9fc9-8eededa156c1

kdorosh commented:
/kick regression test flake https://console.cloud.google.com/cloud-build/builds/7f6939f4-c0c1-41ac-b39f-b05d40c3c175?project=solo-public

kdorosh commented:
/test cluster was in bad state, cleaned up since https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=159010ed-45c8-4f2c-b6bb-75b503e1e74d

yuval-k commented on install/helm/gloo/templates/8-gateway-proxy-service.yaml:
```diff
@@ -27,19 +27,31 @@ spec:
     targetPort: {{ $spec.podTemplate.httpsPort }}
     protocol: TCP
     name: https
+    {{- if and (eq $spec.service.type "NodePort") $spec.service.httpsNodePort }}
```
load balancer services also have a nodeport

kdorosh commented:
/skip-ci
